internal/ingress/backend/endpoint.go

```diff
@@ -54,6 +58,55 @@ func (resolver *endpointResolver) Resolve(ingress *extensions.Ingress, backend *
 	return resolver.resolveIP(ingress, backend)
 }
 
+// For each item in the targets slice, returns the corresponding pod in the result slice at the same index. The result slice is exactly as long as the input slice.
+func (resolver *endpointResolver) ReverseResolve(ingress *extensions.Ingress, backend *extensions.IngressBackend, targets []*elbv2.TargetDescription) ([]*corev1.Pod, error) {
+	service, servicePort, err := findServiceAndPort(resolver.store, ingress.Namespace, backend.ServiceName, backend.ServicePort)
+	if err != nil {
+		return nil, err
+	}
+	serviceKey := ingress.Namespace + "/" + service.Name
+	eps, err := resolver.store.GetServiceEndpoints(serviceKey)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to find service endpoints for %s: %v", serviceKey, err.Error())
+	}
+
+	podMap := map[string]*corev1.Pod{}
+	pods := resolver.store.GetServicePods(service.Spec.Selector)
+	for _, pod := range pods {
+		podMap[pod.Name] = pod
+	}
+
+	targetsMap := map[string]int{}
+	for i, target := range targets {
+		targetsMap[aws.StringValue(target.Id)] = i
+	}
+
+	result := make([]*corev1.Pod, len(targets))
+	for _, epSubset := range eps.Subsets {
+		for _, epPort := range epSubset.Ports {
+			// servicePort.Name is optional if there is only one port
+			if servicePort.Name != "" && servicePort.Name != epPort.Name {
+				continue
+			}
+			for _, epAddr := range append(epSubset.Addresses, epSubset.NotReadyAddresses...) {
+				if epAddr.TargetRef == nil || epAddr.TargetRef.Kind != "Pod" {
+					continue
+				}
+
+				pod, ok := podMap[epAddr.TargetRef.Name]
```

Comment:
why we don't get to pod directly using store(or better controller-runtime client/cache) directly. It will be in-memory lookup and avoid the dependency of service selector.

Reply:
Sounds good, will change it accordingly.